### PR TITLE
Add flip switch to SurfaceTool.generate_normals

### LINKED ISSUE #16528: SurfaceTool::generate_normals are flipped
**Godot version:**

3.0

**OS/device including version:**

Mac OS 10.13.3

**Issue description:**

While making geometry with the Surface Tool, the normals are in the wrong direction. Given the CCW windings in this script and godot's RHCS, the normals should be in the other direction and coincide with the z axis (x cross y = z).

**Steps to reproduce:**

```
extends MeshInstance

func _ready():
  var surfTool = SurfaceTool.new()
  
  surfTool.begin(Mesh.PRIMITIVE_TRIANGLES)
  
  surfTool.add_vertex(Vector3(-1,-1,0))
  surfTool.add_vertex(Vector3( 1,-1,0))
  surfTool.add_vertex(Vector3( 1, 1,0))
  surfTool.add_vertex(Vector3(-1, 1,0))

  surfTool.add_index(0);  
  surfTool.add_index(1);  
  surfTool.add_index(2);  
  
  surfTool.add_index(2);  
  surfTool.add_index(3);  
  surfTool.add_index(0);  
  
  surfTool.generate_normals()

  var material = SpatialMaterial.new()
  material.set_albedo(Color(0,1,0,1))
  
  surfTool.set_material(material)
  
  var mesh = surfTool.commit()

  set_mesh(mesh)
	
```

**Minimal reproduction project:**

[temp.zip](https://github.com/godotengine/godot/files/1713060/temp.zip)

### PR DESCRIPTION
**Pull request by Chaosus**
The SurfaceTool.generate_normals definitly must have a flip switch, of course it's false by default, so it does not break existing code. Closes #16528

### CODE REVIEW

**Comment by Chaosus** [comment]
Its not a bug, its a more like enchantment..

**Comment by Chaosus** [comment]
Ok, i've added missed docs, its complete correct now :) Waiting for merge..

File: doc/classes/SurfaceTool.xml [hunk 1]
```diff
@@ -196,8 +196,11 @@
 		<method name="generate_normals">
 			<return type="void">
 			</return>
+			<argument index="0" name="flip" type="bool" default="false">
+			</argument>
 			<description>
 				Generates normals from Vertices so you do not have to do it manually.
+				Setting "flip" to [code]true[/code] inverts resulted normals.
```
Comment (akien-mga) [inline doc/classes/SurfaceTool.xml]: It should be "result**ing** normals".

Comment (Chaosus) [reply doc/classes/SurfaceTool.xml]: Done !

**Comment by Chaosus** [comment]
@akien-mga Is it correct now ?

**Pull request merged**